src/algorithms/reco/JetReconstruction.cc

```diff
@@ -4,26 +4,44 @@
 // class definition
 #include "JetReconstruction.h"
 
-// standard c includes
+// c utilities
 #include <cmath>
 // event data model related classes
 #include <edm4eic/vector_utils.h>
 #include <edm4eic/MutableReconstructedParticle.h>
-// fastjet includes
+// for fastjet objects
 #include <fastjet/PseudoJet.hh>
 #include <fastjet/ClusterSequenceArea.hh>
 
 using namespace fastjet;
 
-
 namespace eicrecon {
 
   void JetReconstruction::init(std::shared_ptr<spdlog::logger> logger) {
+
     m_log = logger;
     m_log->trace("Initialized");
+
+    // if specified algorithm, recomb. scheme, or area type
+    // are not defined, then issue warning and set it to
+    // default values
+    if (m_mapJetAlgo.find(m_cfg.jetAlgo) == m_mapJetAlgo.end()) {
+      m_log->warn(" Unknown jet algorithm \"{}\" specified! Setting algorithm to default and proceeding.", m_cfg.jetAlgo);
+      m_cfg.jetAlgo = m_defaultFastjetOpts.jetAlgo;
+    }
+    if (m_mapRecombScheme.find(m_cfg.recombScheme) == m_mapRecombScheme.end()) {
+      m_log->warn(" Unknown recombination scheme \"{}\" specified! Setting scheme to default and proceeding.", m_cfg.recombScheme);
+      m_cfg.recombScheme = m_defaultFastjetOpts.recombScheme;
+    }
+    if (m_mapAreaType.find(m_cfg.areaType) == m_mapAreaType.end()) {
+      m_log->warn(" Unknown area type \"{}\" specified! Setting type to default and proceeding.", m_cfg.areaType);
+      m_cfg.areaType = m_defaultFastjetOpts.areaType;
+    }
```

Comment:
If user's choice of option is not found, don't replace it with something else. Throw an exception (we commonly use JException), let it complain loudly.

Reply:
On 2nd thought: I like that approach. My 1st thought was to keep it "not so intrusive", but if a user is actively specifying something other than the default configuration, then they probably want loud complaints. I'll swap out the warns for errors and have it throw an exception!